### PR TITLE
feat: breadcrumb in Refine options

### LINKED ISSUE #2816: [FEAT] Breadcrumb options support
### Is your feature request related to a problem? Please describe.

CRUD components import the `Breadcrumb`([example Ant Design](https://refine.dev/docs/api-reference/antd/components/breadcrumb/)) component by default. If we want to hide the breadcrumbs, we must set the `breadcrumb` property of the CRUD components to `false`.
```ts
<Create breadcrumb={false} />
```
example: https://refine.dev/docs/api-reference/antd/components/basic-views/create/#breadcrumb

instead, the `breadcrumb` property can be added to the `options` of the [`<Refine />`](https://refine.dev/docs/api-reference/core/components/refine-config/#options-1) component. Thus it is managed globally. The `breadcrumb` property passed to the CRUD component overrides the `options`.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Describe the thing to improve

Breadcrumbs in CRUD components are more manageable

### PR DESCRIPTION
**Pull request by TDP17**
As discussed in the issue #2816 - added breadcrumb to Refine options that ends up in RefineContext. Accessed this context in each individual CRUD component. 
**Option passed in individual CRUD component takes priority over the option in RefineContext**

### Test plan (required)

- [x] Automated tests 
image
- [ ] Manually tested via live-examples - option precedence of crud over refine, edge cases like undefined and component

### Closing issues
Closes #2816

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed

To be done
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [ ] Changesets are provided or not needed

### CODE REVIEW

**Comment by aliemir** [comment]
Hey @TDP17, thank you for your effort! I know this is WIP but wanted to give a quick feedback for the lines;

```jsx
const breadcrumb = breadcrumbFromProps ?? options?.breadcrumb;
```

This will be breaking change for users hiding breadcrumb component in the current versions (by passing `breadcrumb={null}`, instead of `??` I think we need to do `typeof` and check for `undefined` 🤔 

We missed this check in our tests 🤦 What do you think? If you come up with a different way, I'll be happy to hear 🚀

**Comment by omeraplak** [comment]
> const breadcrumb: ReactNode =
>   typeof breadcrumbFromProps === "undefined"
>       ? options?.breadcrumb
>       : breadcrumbFromProps;

I agree. Ternary LGTM! 🎃 
Ternary

**Comment by omeraplak** [comment]
> * 3

Hey @TDP17 ,
Thank you so much for your great contribution and hard work! looks great at first glance. We'll probably merge it today.

As a suggestion, a [`tip`](https://docusaurus.io/docs/next/markdown-features/admonitions) can be added to the `breadcrumb` [*](https://refine.dev/docs/api-reference/mui/components/basic-views/create/#breadcrumb) [**](https://refine.dev/docs/api-reference/mantine/components/basic-views/create/#breadcrumb) [***](https://refine.dev/docs/api-reference/antd/components/breadcrumb/) section of CRUD components. that this feature can be managed globally

You can have a look at the past releases for the changeset, https://github.com/refinedev/refine/releases

As an example of this feature,

---
Added ability to manage breadcrumb component globally via `options`
## Usage

```tsx
<Refine
options= {{
   breadcrumb: false, // hide globally
}}
```
or
```tsx
<Refine
options= {{
   breadcrumb: <MyCustomBreadcrumbComponent /> // custom component
}}
```
---

File: packages/mantine/src/components/crud/create/index.tsx [hunk 1]
```diff
@@ -64,6 +65,20 @@ export const Create: React.FC<CreateProps> = (props) => {
...
+    const { options } = useRefineContext();
+
+    const breadcrumb: ReactNode =
+        typeof breadcrumbFromProps === "undefined"
+            ? options?.breadcrumb
+            : breadcrumbFromProps;
+
+    const breadcrumbComponent =
+        typeof breadcrumb !== "undefined" ? (
+            <>{breadcrumb}</> ?? undefined
```
Comment (salihozdemir) [inline packages/mantine/src/components/crud/create/index.tsx:77]:
```suggestion
            {breadcrumb}
```

Isn't the undefined check already done above?

File: packages/mantine/src/components/crud/edit/index.tsx [hunk 2]
```diff
@@ -88,6 +89,19 @@ export const Edit: React.FC<EditProps> = (props) => {
...
 
+    const { options } = useRefineContext();
+    const breadcrumb: ReactNode =
+        typeof breadcrumbFromProps === "undefined"
+            ? options?.breadcrumb
+            : breadcrumbFromProps;
+
+    const breadcrumbComponent =
+        typeof breadcrumb !== "undefined" ? (
+            <>{breadcrumb}</> ?? undefined
```
Comment (salihozdemir) [inline packages/mantine/src/components/crud/edit/index.tsx:100]:
```suggestion
            {breadcrumb}
```

File: packages/mantine/src/components/crud/list/index.tsx [hunk 3]
```diff
@@ -64,6 +65,19 @@ export const List: React.FC<ListProps> = (props) => {
...
 
+    const { options } = useRefineContext();
+    const breadcrumb: ReactNode =
+        typeof breadcrumbFromProps === "undefined"
+            ? options?.breadcrumb
+            : breadcrumbFromProps;
+
+    const breadcrumbComponent =
+        typeof breadcrumb !== "undefined" ? (
+            <>{breadcrumb}</> ?? undefined
```
Comment (salihozdemir) [inline packages/mantine/src/components/crud/list/index.tsx:76]:
```suggestion
            {breadcrumb}
```

File: packages/mantine/src/components/crud/show/index.tsx [hunk 4]
```diff
@@ -78,6 +79,19 @@ export const Show: React.FC<ShowProps> = (props) => {
...
 
+    const { options } = useRefineContext();
+    const breadcrumb: ReactNode =
+        typeof breadcrumbFromProps === "undefined"
+            ? options?.breadcrumb
+            : breadcrumbFromProps;
+
+    const breadcrumbComponent =
+        typeof breadcrumb !== "undefined" ? (
+            <>{breadcrumb}</> ?? undefined
```
Comment (salihozdemir) [inline packages/mantine/src/components/crud/show/index.tsx:90]:
```suggestion
            {breadcrumb}
```

**Review by salihozdemir — CHANGES_REQUESTED** [review]
Thank you four your effort 🙏🏻.
I leave some review comments if you can fix them, I'll merge this great PR 🚀

In addition, since the reviews I added are the same as the other parts you made changes, I did not write them one by one. 🤙🏻

**Comment by aliemir** [comment]
Hey @TDP17 thanks for the PR and the latest changes. @salihozdemir dropped some minor change requests, can you check those? Looks like we can infer the type from the definition rather than stating it there. 

For the changesets, you can just run `npm run changeset` command to create one with the selected packages. We prefer separate changesets for every package but in this case we can just create two; one for the `refine-core` and the other is for UI packages. (You can mark those changes as `minor`) 

For more info on how to write descriptions for the changesets, you can check out previous PRs or the changelog for the releases 🚀 

(I prefer writing them inside the code editor by editing the markdown file created by the command rather than using command line) 

After these are done, we're ready for the merge and release 🚀 🚀

**Comment by omeraplak** [comment]
> One small bug I ran into, running `npm run changeset` gives me the following error
> 
> ```
> 🦋  error Error: Failed to find where HEAD diverged from master. Does master exist?
> ```
> 
> I fixed this by changing `.changeset/config.json` > `"baseBranch": "master",` to ` "baseBranch": "next",` Should I add this fix in this PR itself? Or is it supposed to be this way?

You may be getting this error because the master branch is not in your local. You can ignore it

File: .changeset/afraid-dolphins-sing.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+Feat: Added ability to manage breadcrumb component globally via options
+
+## Usage
+```jsx
+<Refine
+options= {{
+   breadcrumb: false, // hide globally
+}}
+```
```
Comment (aliemir) [inline .changeset/afraid-dolphins-sing.md:13]:
```suggestion
```jsx
<Refine
    options= {{
        breadcrumb: false, // hide globally
    }}
/>
```
```

File: .changeset/afraid-dolphins-sing.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+or
+```jsx
+<Refine
+options= {{
+   breadcrumb: <MyCustomBreadcrumbComponent /> // custom component
+}}
```
Comment (aliemir) [inline .changeset/afraid-dolphins-sing.md:19]:
```suggestion
<Refine
    options= {{
        breadcrumb: <MyCustomBreadcrumbComponent /> // custom component
    }}
/>
```

**Review by aliemir — APPROVED** [review]
Thank you!

**Review by omeraplak — APPROVED** [review]
